src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1677,20 +1677,19 @@ ucs_status_t uct_dc_mlx5_ep_check_fc(uct_dc_mlx5_iface_t *iface,
          * uct_dc_mlx5_fc_remove_ep() can rely on "seq != UINT64_MAX" condition
          * to check that a FC entry's sequnce number needs to be verified */
         ucs_assert(iface->tx.fc_seq != UINT64_MAX);
-        fc_entry->seq = iface->tx.fc_seq++;
+        fc_entry->seq = iface->tx.fc_seq;
     }
 
     fc_entry->send_time = now;
     status              = uct_dc_mlx5_ep_fc_hard_req_send(ep, fc_entry->seq);
     if (status != UCS_OK) {
         if (ret != UCS_KH_PUT_KEY_PRESENT) {
-            /* Decrement a FC sequence number to roll back changes done above,
-             * because FC_HARD_REQ wasn't sent with this sequence number */
-            iface->tx.fc_seq--;
             kh_del(uct_dc_mlx5_fc_hash, &iface->tx.fc_hash, it);
         }
 
         goto out;
+    } else if (ret == UCS_KH_PUT_KEY_PRESENT) {
+            iface->tx.fc_seq++;
```

Comment:
- indent
- why else if have goto?
- maybe ++ without checking ret?

Reply:
> indent
> why else if have goto?

done

> maybe ++ without checking ret?

I guess that better to increment only if FC seq was really used

Reply:
ok, removed